packages/account-tree-controller/src/AccountTreeController.ts

```diff
@@ -206,6 +214,42 @@ export class AccountTreeController extends BaseController<
     }
   }
 
+  #getRuleForWallet<WalletType extends AccountWalletType>(
+    wallet: AccountWalletObjectOf<WalletType>,
+  ): Rule<WalletType, AccountGroupType> {
+    switch (wallet.type) {
+      case AccountWalletType.Entropy:
+        return this.#getEntropyRule() as unknown as Rule<
+          WalletType,
+          AccountGroupType
+        >;
+      case AccountWalletType.Snap:
+        return this.#getSnapRule() as unknown as Rule<
+          WalletType,
+          AccountGroupType
+        >;
+      default:
+        return this.#getKeyringRule() as unknown as Rule<
+          WalletType,
+          AccountGroupType
+        >;
+    }
+  }
+
+  #isNewGroup(group: AccountGroupObject): boolean {
+    // Check if any account in the group was created after service start
+    for (const id of group.accounts) {
+      const account = this.messagingSystem.call(
+        'AccountsController:getAccount',
+        id,
+      );
+      if (account && account.metadata.importTime > this.#serviceStartTime) {
+        return true;
+      }
+    }
+    return false;
+  }
+
```

Comment:
We should probably just add that piece of information when the group is created in the `match` method of each rule instead of doing repeated calls for each account again.